src/crush/CrushWrapper.cc

```diff
@@ -2673,7 +2677,7 @@ int CrushWrapper::device_class_clone(
   for (auto& w : choose_args) {
     crush_choose_arg_map& cmap = w.second;
     if (-1-bno >= (int)cmap.size) {
```

Comment:
Shouldn't this be `if ((int)cmap.size < crush->max_buckets)` too?

Reply:
yes!